File: app/src/main/java/com/ngengs/android/popularmovies/apps/data/local/MoviesTopRated.java
package com.ngengs.android.popularmovies.apps.data.local;

import android.content.ContentResolver;
import android.net.Uri;
import android.provider.BaseColumns;

import com.ngengs.android.popularmovies.apps.globals.LocalDatabase;

/**
 * Created by ngengs on 7/3/2017.
 */

@SuppressWarnings({"WeakerAccess", "unused"})
public final class MoviesTopRated implements BaseColumns {
    public static final Uri CONTENT_URI = MoviesEntry.CONTENT_URI.buildUpon()
            .appendPath(LocalDatabase.PATH_TOP_RATED)
            .build();
    public static final String CONTENT_DIR_TYPE =
            ContentResolver.CURSOR_DIR_BASE_TYPE + "/" + LocalDatabase.CONTENT_AUTHORITY + "/" + LocalDatabase.PATH_MOVIES
                    + "/" + LocalDatabase.PATH_TOP_RATED;

    public static final String TABLE_NAME = "top_rated_movies";

    public static final String SQL_CREATE_TABLE =
            "CREATE TABLE " + TABLE_NAME + " (" +
                    _ID + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
                    LocalDatabase.COLUMN_MOVIE_ID_KEY + " INTEGER NOT NULL, " +

                    " FOREIGN KEY (" + LocalDatabase.COLUMN_MOVIE_ID_KEY + ") REFERENCES " +
                    MoviesEntry.TABLE_NAME + " (" + MoviesEntry._ID + ") " +

                    " );";

    private static final String[] COLUMNS = {_ID, LocalDatabase.COLUMN_MOVIE_ID_KEY};

    public static String[] getColumns() {
        return COLUMNS.clone();
    }
}

File: app/src/main/java/com/ngengs/android/popularmovies/apps/data/local/MoviesTopRated.kt
package com.ngengs.android.popularmovies.apps.data.local

import android.provider.BaseColumns
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.ngengs.android.popularmovies.apps.globals.LocalDatabase

/**
 * Created by rizky.kharisma on 10/11/22.
 * @ngengs
 */
@Entity(
    tableName = LocalDatabase.PATH_TOP_RATED,
    foreignKeys = [
        ForeignKey(
            entity = Movies::class,
            parentColumns = arrayOf(BaseColumns._ID),
            childColumns = arrayOf(LocalDatabase.COLUMN_MOVIE_ID_KEY),
        )
    ]
)
data class MoviesTopRated(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = BaseColumns._ID)
    val id: Int? = null,
    @ColumnInfo(name = LocalDatabase.COLUMN_MOVIE_ID_KEY, index = true)
    val movieId: Int,
)
